src/processing_pipeline/stage_3.py

```
@@ -246,66 +254,203 @@ class Stage3Executor:
+                tools=[Tool(google_search=GoogleSearch())],
+                thinking_config=ThinkingConfig(thinking_budget=4096),
+                safety_settings=cls.__get_safety_settings(),
+            ),
+        )
+
+        grounding_metadata = str(response.candidates[0].grounding_metadata) if response.candidates else None
+
+        if not response.text:
+            finish_reason = response.candidates[0].finish_reason
+            if finish_reason == FinishReason.MAX_TOKENS:
+                raise ValueError("The response from Gemini was too long and was cut off in step 1.")
```

Ruff (0.13.3)

Avoid specifying long messages outside the exception class

(TRY003)